app/src/app/admin/sales/salesinvoice/salesinvoice.component.ts

```diff
@@ -87,6 +95,105 @@ export class SalesinvoiceComponent {
     this.showSaleInvoiceList = true;
   }
 
+  showPendingOrdersList() {
+    console.log("We are selecting customer here : ");
+    const selectedCustomerId = this.formConfig.model.sale_invoice_order.customer_id;
+    const selectedCustomerName = this.formConfig.model.sale_invoice_order.customer?.name;
+
+    if (!selectedCustomerId) {
+      this.noOrdersMessage = 'Please select a customer.';
+      this.customerOrders = [];
+      this.openModal();
+      return;
+    }
+
+    this.customerOrders = [];
+    this.noOrdersMessage = '';
+
+    this.getPendingOrdersByCustomer(selectedCustomerId).pipe(
+      switchMap(orders => {
+        if (orders.count === 0) {
+          this.noOrdersMessage = `No Pending orders for ${selectedCustomerName}.`;
+          this.customerOrders = [];
+          this.openModal();
+          return [];
+        }
+
+        const detailedOrderRequests = orders.data.map(order =>
+          this.getOrderDetails(order.sale_order_id).pipe(
+            tap(orderDetails => {
+              order.productsList = orderDetails.data.sale_order_items.map(item => ({
+                product_name: item.product?.name ?? 'Unknown Product',
+                quantity: item.quantity
+              }));
+            })
+          )
+        );
+
+        return forkJoin(detailedOrderRequests).pipe(
+          tap(() => {
+            this.customerOrders = orders.data;
+            this.openModal();
+          })
+        );
+      })
+    ).subscribe();
+  }
+
+  openModal() {
+    this.ordersModal.nativeElement.classList.add('show');
+    this.ordersModal.nativeElement.style.display = 'block';
+    document.body.classList.add('modal-open');
+    document.body.style.overflow = 'hidden';
+  }
+
+  hideModal() {
+    const modalBackdrop = document.querySelector('.modal-backdrop');
+    if (modalBackdrop) {
+      modalBackdrop.remove();
+    }
+    this.ordersModal.nativeElement.classList.remove('show');
+    this.ordersModal.nativeElement.style.display = 'none';
+    document.body.classList.remove('modal-open');
+    document.body.style.overflow = '';
+  }
+
+  getPendingOrdersByCustomer(customerId: string): Observable<any> {
+    const url = `sales/sale_order_search/?customer_id=${customerId}&order_status_id=085266c9-5020-41b3-ab58-1e4d88f4ff19`;
```

Comment:
Why the order_status_id is hardcoded here? Please change

Reply:
done